docs/contributor/arch/013-resource-attributes-enrichment-for-non-workload-metrics.md

```diff
@@ -0,0 +1,61 @@
+# 2. Resource Attributes Enrichment for Non-Workload Metrics
+
+Date: 2024-08-30
+
+## Status
+
+Accepted
+
+## Context
+
+Currently, the [k8sattributes processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/k8sattributesprocessor/README.md) is used in the Metric Gateway to enrich metrics resource attributes with Kubernetes metadata.
+This is very useful for workload metrics (like Pod and container metrics). However, for non-workload metrics (like Node metrics), the metrics will be incorrectly associated with Metric Agent Pod, because the Metric Agent Pod is the one that emits these metrics.
+
+For example, in the diagram shown below, the `k8s.node.cpu.usage` metric is incorrectly associated with the `telemetry-metric-agent-4ncx9` pod. This creates confusion, as the user might incorrectly think that this metric provides the amount of node CPU consumed by the `telemetry-metric-agent-4ncx9` pod. While in reality, it provides the entire node CPU usage.
+In addition, the metrics from the system namespaces (including `kyma-system` namespace) are excluded by default. Thus, this metric is dropped by default even if the user enables the collection of Node metrics.
+
+![Node Metric With k8sattributes Processor](../assets/node-metric-with-k8sattributes-processor.png)
+
+
+## Decision
+
+There are different possible solutions to solve this problem:
+
+### Option 1: Using Connectors
+
+![Connectors](../assets/connectors.drawio.svg)
+
+We can split our MetricPipeline into 3 sub-pipelines which are connected using [Connectors](https://opentelemetry.io/docs/collector/building/connector/) as shown in the diagram above.
+A `Receiver Pipeline`, which contains the receivers components.
+A `k8sattribute Processor Pipeline` which contains only the k8sattributes processor.
+A `Processor + Exporter Pipeline` which contains the rest of the processors and the exporters. The metrics will be routed to the `k8sattribute Processor Pipeline` only if they are workload metrics.
+Otherwise, the metrics will bypass the `k8sattribute Processor Pipeline` and will be routed directly to the `Processor + Exporter Pipeline`.
+
+- _Pro_: Clean solution. The non-workload metrics will never have the unwanted resource attributes set to any value.
+- _Con_: Complex setup.
```

Comment:
Per MetricPIpeline entity we will have 3 pipelines in the collector instead of 1. The set of processors is probably the same, so we will have a more complex setup of the pipeline definitions in combination with the new connector definitions. Additionally, operation will get more challenging as in the metrics you cannot map directly a collector pipeline to a MetricPipeline.